web/packages/api/src/status.ts

```diff
@@ -1,58 +1,120 @@
-import { IERC20Metadata__factory, IERC20__factory } from '@snowbridge/contract-types'
-import { Context } from './index'
-import { fetchBeaconSlot } from './utils'
+import { IERC20Metadata__factory, IERC20__factory } from "@snowbridge/contract-types"
+import { Context } from "./index"
+import { fetchBeaconSlot } from "./utils"
 
-export type OperatingMode = 'Normal' | 'Halted'
+export type OperatingMode = "Normal" | "Halted"
 export type BridgeStatusInfo = {
     toEthereum: {
         operatingMode: {
-            outbound: OperatingMode,
-        },
-        latestPolkadotBlockOnEthereum: number,
-        latestPolkaotBlock: number,
-        blockLatency: number,
-        latencySeconds: number,
-    },
+            outbound: OperatingMode
+        }
+        latestPolkadotBlockOnEthereum: number
+        previousPolkadotBlockOnEthereum: number
+        latestPolkaotBlock: number
+        blockLatency: number
+        latencySeconds: number
+    }
     toPolkadot: {
         operatingMode: {
-            beacon: OperatingMode,
-            inbound: OperatingMode,
-            outbound: OperatingMode,
-        },
-        latestEthereumBlockOnPolkadot: number,
-        latestEthereumBlock: number,
-        blockLatency: number,
-        latencySeconds: number,
-    },
+            beacon: OperatingMode
+            inbound: OperatingMode
+            outbound: OperatingMode
+        }
+        latestEthereumBlockOnPolkadot: number
+        previousEthereumBlockOnPolkadot: number
+        latestEthereumBlock: number
+        blockLatency: number
```

Comment:
You can pass `blockLatency` to channelStatusInfo to get more accurate results instead of relying on the defaults.

Reply:
The basic idea is just to read storage at a history block(interval long enough) so to compare with the latest. We can later monitor with alarm rules something like `event X happens Y times in a time window Z`, then most likely the channel is stale and something error.